src/components/scopedVariables/SavedVariables.tsx

```diff
@@ -1,20 +1,41 @@
-import React, { useState, useRef } from 'react'
+import React, { useState, useRef, useEffect } from 'react'
 import { LoadScopedVariables, ScopedVariablesEditor } from './UploadScopedVariables'
+import { TableList, TableItem } from './Table'
 import { useFileReader, useClickOutside } from './utils/hooks'
 import CodeEditor from '../CodeEditor/CodeEditor'
 import Descriptor from './Descriptor'
 import { downloadData } from './utils/helpers'
-import { FileView, SavedVariablesViewI } from './types'
-import { DOWNLOAD_FILE_NAME, DOWNLOAD_TEMPLATE_NAME, DROPDOWN_ITEMS, VARIABLES_TEMPLATE } from './constants'
+import { FileView, SavedVariablesViewI, VariableListItemI } from './types'
+import {
+    DOWNLOAD_FILE_NAME,
+    DOWNLOAD_TEMPLATE_NAME,
+    DROPDOWN_ITEMS,
+    VARIABLES_TEMPLATE,
+    SAMPLE_DATA,
+    TABLE_LIST_HEADINGS,
+} from './constants'
 import { ReactComponent as ICFileDownload } from '../../assets/icons/ic-file-download.svg'
 
 const SavedVariablesView = ({ scopedVariables, setScopedVariables }: SavedVariablesViewI) => {
     const [showDropdown, setShowDropdown] = useState<boolean>(false)
     const [currentView, setCurrentView] = useState<FileView>(FileView.YAML)
+    const [variablesList, setVariablesList] = useState<VariableListItemI[]>(null)
     const dropdownRef = useRef(null)
 
     const { status, progress, fileData, abortRead, readFile } = useFileReader()
 
+    useEffect(() => {
+        if (status?.status == null) {
```

Comment:
or Can we use  !status?.status

Reply:
It is a boolean so the status false would mean error state